### DISCUSSION TITLE
Support for quickly toggling Opt between Meta and character modifier?

### DISCUSSION THREAD

**Discussion by nico** [ghostty-org/ghostty#2517] [Feature Requests, Ideas] [answered]
One Terminal.app feature that i love is Cmd-Opt-O for toggling quickly toggling the behavior of the "Opt" key on mac keyboards between opt-f inserting ƒ and opt-f sending meta-f for going forward a word (…depending on the shell i guess, but bash and zsh both do that). This is not specific to `f` of course, it toggles the behavior of the Opt key generally.

In Terminal.app, there's a "Use Option as Meta Key" menu entry in the Edit menu. It defaults to unchecked, and Cmd-Opt-O toggles it.

In iTerm, this is instead in Preferences->Profiles->Keys, which has options for Left Option key, and Right Option key.

Personally, I like Terminal.app's approach a bit more, since a) I change this fairly often (several times / day), depending on if I want to enter things like `µ`, or if I want to use esc-f to go forward a word, or esc-d to delete-forward b) having different behavior for left and right option keys seems complicated.

Is this something you'd accept a patch for?

**Comment by mitchellh** [comment]
I like this idea. I think introducing a new binding action such as `toggle_macos_option_as_alt` would be perfectly fine. Do you know if this affects both sides (left and right)?

**Reply by nico** [reply]
In Terminal.app it affects both the left and right opt key, yes.

Forgot to mention, there's some UI feedback when toggling it. Here's me rapidly toggling it back and forth:

https://github.com/user-attachments/assets/972a78c0-f1cd-4dad-803b-8b6971f06456

**Reply by mitchellh** [reply]
Got it thanks. We can separate the UI feedback from the binding to simplify an initial PR. 

Given what you said, I think it's fair to introduce a new binding action `toggle_macos_option_as_alt` and have it swap the value between `false` and `true` (both sides). Once a binding action is introduced, it's trivial to bind the state to the menu item. ``

**Reply by mitchellh** [reply, marked as answer]
I made an issue #2520 to track this well scoped feature. Thank you!
